php-80/Dockerfile

```diff
@@ -71,35 +71,25 @@ RUN mkdir -p ${BUILD_DIR}  \
 
 ###############################################################################
 # ZLIB Build
+# We compile a newer version because Lambda uses an old version (1.2.7) that
+# has a security vulnerability (CVE-2022-37434).
+# See https://github.com/brefphp/aws-lambda-layers/pull/110
+# Can be removed once Lambda updates their version.
 # https://github.com/madler/zlib/releases
-# Needed for:
-#   - openssl
-#   - curl
-#   - php
-# Used By:
-#   - xml2
 ENV VERSION_ZLIB=1.3
-ENV ZLIB_BUILD_DIR=${BUILD_DIR}/xml2
-
+ENV ZLIB_BUILD_DIR=${BUILD_DIR}/zlib
 RUN set -xe; \
     mkdir -p ${ZLIB_BUILD_DIR}; \
-# Download and upack the source code
-    curl -Ls http://zlib.net/zlib-${VERSION_ZLIB}.tar.xz \
-  | tar xJC ${ZLIB_BUILD_DIR} --strip-components=1
-
-# Move into the unpackaged code directory
+    curl -Ls https://github.com/madler/zlib/releases/download/v${VERSION_ZLIB}/zlib-${VERSION_ZLIB}.tar.gz \
+  | tar xzC ${ZLIB_BUILD_DIR} --strip-components=1
 WORKDIR ${ZLIB_BUILD_DIR}/
-
-# Configure the build
 RUN set -xe; \
     make distclean \
  && CFLAGS="" \
     CPPFLAGS="-I${INSTALL_DIR}/include  -I/usr/include" \
     LDFLAGS="-L${INSTALL_DIR}/lib64 -L${INSTALL_DIR}/lib" \
     ./configure \
-    --prefix=${INSTALL_DIR} \
-    --64
```

Comment:
@GrahamCampbell I removed this line (after reading https://github.com/madler/zlib/issues/143) at it seemed to do the trick. Do you see any reason to be worried?

If not I'll merge and release.

Reply:
Ah yeh, I think i did something similar when making vapor work with arm. This is fine. 👍